### PR TITLE
Closes #15 align cdc metadata solution with who

### LINKED ISSUE #15: General Issue: Recreate CDC growth-chart data using code
### Background Information

From discussion #13:

```
If anyone wants to re-create the reference files, the reference data for the CDC charts are at
https://www.cdc.gov/growthcharts/percentile_data_files.htm
You'd only need the data in numbers 6 (wt), ht (7), and 8 (BMI). You only need that data for sex, age, L, M, and S. But having some of the calculated percentiles might be a good check on the calculations.
[L M S are the 3 smoothing parameters to transform the data to approximate a normal distribution - they're the L (Box-Cox transformation for normality), M (median), and S (coefficient of variation)
```

Recreate the data using `tribble()` for replicability as data source may change overtime and its easier to track than floating csv's
Use snapshots for testthat 

### Definition of Done

- [ ] Render each necessary CDC dataset as specified above into an .rda object using `tibble::tribble()`
- [ ] Use `testthat::expect_snapshot()` so we can track changes in the feature

### PR DESCRIPTION
**Pull request by zdz2101**
Thank you for your Pull Request! We have developed this task checklist from the [Development Process Guide](https://pharmaverse.github.io/admiral/CONTRIBUTING.html#detailed-development-process) to help with the final steps of the process. Completing the below tasks helps to ensure our reviewers can maximize their time on your code as well as making sure the admiral codebase remains robust and consistent.   

Please check off each taskbox as an acknowledgment that you completed the task or check off that it is not relevant to your Pull Request. This checklist is part of the Github Action workflows and the Pull Request will not be merged into the `main` branch until you have checked off each task.

- [x] Place Closes #<insert_issue_number> into the beginning of your Pull Request Title (Use Edit button in top-right if you need to update)
- [x] Code is formatted according to the [tidyverse style guide](https://style.tidyverse.org/). Run `styler::style_file()` to style R and Rmd files
- [x] Updated relevant unit tests or have written new unit tests, which should consider realistic data scenarios and edge cases, e.g. empty datasets, errors, boundary cases etc. - See [Unit Test Guide](https://pharmaverse.github.io/admiraldev/articles/unit_test_guidance.html#tests-should-be-robust-to-cover-realistic-data-scenarios)
- [x] If you removed/replaced any function and/or function parameters, did you fully follow the [deprecation guidance](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#deprecation)?
- [x] Update to all relevant roxygen headers and examples, including keywords and families. Refer to the [categorization of functions](https://pharmaverse.github.io/admiraldev/articles/programming_strategy.html#categorization-of-functions) to tag appropriate keyword/family.
- [x] Run `devtools::document()` so all `.Rd` files in the `man` folder and the `NAMESPACE` file in the project root are updated appropriately
- [x] Address any updates needed for vignettes and/or templates
- [x] Update `NEWS.md` under the header `# admiralpeds (development version)` if the changes pertain to a user-facing function (i.e. it has an `@export` tag) or documentation aimed at users (rather than developers)
- [x] Build admiralpeds site `pkgdown::build_site()` and check that all affected examples are displayed correctly and that all new functions occur on the Reference page. 
- [x] Address or fix all lintr warnings and errors - `lintr::lint_package()`
- [x] Run `R CMD check` locally and address all errors and warnings - `devtools::check()`
- [x] Link the issue in the Development Section on the right hand side.
- [x] Address all merge conflicts and resolve appropriately
- [x] Pat yourself on the back for a job well done! Much love to your accomplishment!

### CODE REVIEW

File: data-raw/growthchart_metadata_creation.R [hunk 1]
```diff
@@ -57,3 +69,8 @@ usethis::use_data(who_bmi_for_age_boys, overwrite = TRUE)
 
 usethis::use_data(who_hc_for_age_girls, overwrite = TRUE)
 usethis::use_data(who_hc_for_age_boys, overwrite = TRUE)
+
+usethis::use_data(cdc_bmiage, overwrite = TRUE)
```
Comment (zdz2101) [inline data-raw/growthchart_metadata_creation.R:73]: @Pierre-Wallet I've mimic'd the way you handled your PR and renamed the metadata creation file to be generalized and contain the cdc datasets too

**Review by Pierre-Wallet — COMMENTED** [review]
There is no @title, @description and @format tags in your documentation.
Not mandatory but it would be more consistent to have it.

I made a comparison between your internal datasets and the source, it is ok for me.

**Review by Pierre-Wallet — APPROVED** [review]

**Comment by rossfarrugia** [comment]
thanks @zdz2101 and @Pierre-Wallet for resolving all the metadata issues! just catching up post holidays and all looks fine. in the vignettes we can use the rda from the package here but then also explain that if ever a company needs to change any metadata then in the `data-raw/` folder all the programs that create these are transparently laid out and they could adapt however they need. looks a nice and clean solution! i only spotted something minor: https://github.com/pharmaverse/admiralpeds/issues/28
